src/Reflection/ReflectionClass.php

```diff
@@ -331,6 +331,18 @@ public function hasProperty($name)
     }
 
     /**
+     * Protected API to add properties at runtime (used in ReflectionObject)
+     *
+     * @param ReflectionProperty $property
+     */
+    protected function addProperty(ReflectionProperty $property)
```

Comment:
@asgrim what about having "cachedProperties" (parent class state) and "runtimeProperties" (child class state)?